### DISCUSSION TITLE
Temperature manometer

### DISCUSSION THREAD

**Discussion by bkemmler** [jomjol/AI-on-the-edge-device#1090] [General] [answered]
Hi, Is there any  way to ready out the current temperature as shown in the pic?

![image](https://user-images.githubusercontent.com/49644093/192386087-e3aad1cb-c5b3-4a9c-81a5-9f3603a9ddae.png)

Most of the time is is not working. Once or twice the right value has been shown.

In other discussion i saw a similar question for an oil tank. But i still have a small hope that there is a easy way :-)

Thanks in advance.

**Comment by caco3** [comment]
Yes, there where discussions before.

I think it should work, you would need your own training data, and some consistency checks would need to be disabled.
Also it might need some post processing as the system as is now expects a pointer to go 360° for 10 units.

**Comment by jomjol** [comment, marked as answer]
Yes - totally agree!

**Comment by pipip** [comment]
@bkemmler: Did you find a solution? I also really want to use this for reading my water temperature in order to detect when the heater does not work. I believe I am stuck at the same point as you (setup so far worked perfectly, thanks for the great docs)
![192_168_1_35_2022-10-08_13-32-36](https://user-images.githubusercontent.com/2663439/194712759-52e400b1-f675-4f71-b7f4-6d2c84633f7e.jpg)

**Reply by bkemmler** [reply]
So far I have not found a simple solution. Sorry

**Reply by bkemmler** [reply]
I found a solution which is working: https://www.elec-cafe.com/use-wemos-d1-mini-and-ds18b20-sensor-monitor-temperature-on-home-assistant/

**Comment by RMadosa** [comment]
Hi
Found this page: [ESP32-CAM and OpenCV to read an analog gauge / dial](https://circuits-ninja.pl/reading-an-indication-from-an-analog-pressure-gauge-using-the-esp32-cam-module-with-an-ov2640-and-opencv-camera/) and also on [Git](https://github.com/bartek-szymanski-szyba/ESP32-projects/tree/master/ESP32-CAM-OpenCV-Gauge-read-via-image-processing)
That looks like it fits the question, but unfortunately not based on this project. But I hope this will help someone.

Is this known to anyone or has someone already copied it?

**Comment by pipip** [comment]
@RMadosa: I have done sth similar.
- Flashed ESP32 with ESPHome (https://esphome.io/components/esp32_camera.html and https://esphome.io/components/esp32_camera_web_server.html)
- Setup opencv docker on home server (so not nicely integrated on the esp32)
- Used this script to fetch image from esp32-cam and read value from gauge https://github.com/pipip/analog-gauge-reader
- In my current setup I have integrated the script in home assistant (via shell_comand and file sensor)
